### PR TITLE
[2.3][Process] Use stream based storage to avoid memory issues

### PR DESCRIPTION
**Pull request by romainneutron**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #17390 |
| License | MIT |

### CODE REVIEW

File: src/Symfony/Component/Process/Process.php [hunk 1]
```diff
@@ -437,17 +434,14 @@ public function getErrorOutput()
...
     {
-        $this->requireProcessIsStarted(__FUNCTION__);
-
-        $data = $this->getErrorOutput();
-
-        $latest = substr($data, $this->incrementalErrorOutputOffset);
-
-        if (false === $latest) {
-            return '';
+        if ($this->outputDisabled) {
```
Comment (Tobion) [inline src/Symfony/Component/Process/Process.php]: undefined. see tests

Comment (romainneutron) [inline src/Symfony/Component/Process/Process.php]: woops, fixed

File: src/Symfony/Component/Process/Process.php [hunk 2]
```diff
@@ -378,7 +378,7 @@ public function getOutput()
 
         $this->readPipes(false, '\\' === DIRECTORY_SEPARATOR ? !$this->processInformation['running'] : true);
 
-        return $this->stdout;
+        return (string) stream_get_contents($this->stdout, -1, 0);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Process/Process.php]: casting to string triggers copy-on-writes, it would be better to use an `if` here

**Comment by romainneutron** [comment]
Comments addressed, PR updated

**Comment by romainneutron** [comment]
I just updated the PR, I reduced the temp stream to 1M at max before switching on fs. I've also changed the streams mode from `a+` to `wb+`

**Comment by fabpot** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @romainneutron.

**Pull request merged**